docs/sdks/java.md

```diff
@@ -31,136 +34,214 @@ There's an optional uber-jar with shaded and relocated guava and failsafe depend
+    return prefabCloudClient.featureFlagClient();
+  }
 
-## Get FeatureFlag
+  @Singleton
+  public ConfigClient configClient(PrefabCloudClient prefabCloudClient) {
+    return prefabCloudClient.configClient();
+  }
+}
+```
+
+## Feature Flags
+
+For boolean flags, you can use the `featureIsOn` convenience method:
 
 ```java
-FeatureFlagClient featureFlagClient = prefabCloudClient.featureFlagClient();
+public class MyClass {
+  // assumes you have setup a singleton
+  @Inject
+  private FeatureFlagClient featureFlagClient;
 
-featureFlagClient.featureIsOn(
-    "features.example-flag",
-    PrefabContext.newBuilder("customer")
-      .put("group", "beta")
-      .build()
-  )
+  public String test(String key){
+    boolean val = featureFlagClient.featureIsOn(key);
+    return "Feature flag value of %s is %s".formatted(key, val);
+  }
+}
 ```
 
-## Get Config
+Feature flags don't have to return just true or false. You can get other data types using `get`:
 
 ```java
-final Optional<Prefab.ConfigValue> configValue = prefabCloudClient.configClient().get("the.key");
-if(configValue.isPresent()){
-  System.out.println(configValue.get().getString());
+public class MyClass {
+  // assumes you have setup a singleton
+  @Inject
+  private FeatureFlagClient featureFlagClient;
+
+  public String test(String key){
+    // highlight-next-line
+    Optional<Prefab.ConfigValue> val = featureFlagClient.get(key);
+    return "Feature flag value of %s is %s".formatted(key, val.orElse("no value found"));
+  }
 }
 ```
 
-## Provide Context
+## Context
 
-To finely-target configuration rule evaluation, we accept contextual information both inline when making a get request
+To finely-target configuration rule evaluation, we accept contextual information globally or request-scoped with the ContextStore which will affect all logging, featureflag and config lookups.
 
 ```java
-prefabCloudClient.configClient().get("the.key", PrefabContext.newBuilder("user")
-                                                    .put("name", "james")
-                                                    .put("tier", "gold")
-                                                    .put("customerMonths", 12)
-                                                    .build()
-                                                 )
+prefabCloudClient.configClient().getContextStore().addContext(PrefabContext.newBuilder("User")
+                        .put("name", user.getName())
+                        .build());
 ```
 
-and globally or request-scoped with the ContextStore which will affect all logging, featureflag and config lookups.
+When global context is set, log levels and feature flags will evaluate in that context. Here are details on setting global context:
+
+<Tabs groupId="lang">
+<TabItem value="micronaut" label="Micronaut">
+
+In your Prefab options, set your Prefab Context store to be your ServerRequestContextStore.
 
 ```java
+options.setContextStore(new ServerRequestContextStore());
```

Comment:
this is very micronaut specific so I don't know if needs the prominence it has here - maybe more of a caution like the ruby one for forking app servers